fronted.go

```diff
@@ -212,6 +310,12 @@ func (f *fronted) RoundTrip(req *http.Request) (*http.Response, error) {
 func (f *fronted) RoundTripHijack(req *http.Request) (*http.Response, net.Conn, error) {
 	op := ops.Begin("fronted_roundtrip")
 	defer op.End()
+	// If the request has a context, use it. Otherwise, create a new one that has a timeout.
+	if req.Context() == nil {
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+		req = req.WithContext(ctx)
+	}
```

Comment:
now the timeout is just based on the request context, which to me seems much cleaner